storage/AggregationOperationState.hpp

```diff
@@ -156,6 +152,29 @@ class AggregationOperationState {
       const CatalogDatabaseLite &database);
 
   /**
+   * @brief Get the number of partitions to be used for initializing the
+   *        aggregation.
+   *
+   * @return The number of partitions to be used for initializing the aggregation.
+   **/
+  std::size_t getNumInitializationPartitions() const;
+
+  /**
+   * @brief Get the number of partitions to be used for finalizing the
+   *        aggregation.
+   *
+   * @return The number of partitions to be used for finalizing the aggregation.
+   **/
+  std::size_t getNumFinalizationPartitions() const;
```

Comment:
I get your point, but for the distributed version, my concern is that we have to schedule the finalize work order with a partition to the same node that executes the initialize work order. Thus, your approach makes it harder to implement in the distributed version.

On the other hand, we could measure the overheads of the initialize work order, and I guess it is related small, compared to the whole aggregation processing time.